.github/workflows/releases.yml

```diff
@@ -0,0 +1,103 @@
+name: Releases workflow
+
+on:
+  create:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+env:
+  GO_VERSION: '1.20.6'
+
+permissions:
+  contents: write
+  deployments: write
+
+jobs:
+  # TODO: Any way we can just call build.yml?
```

Comment:
Technically possible, but no sure if there are trade-offs / effort required.
https://docs.github.com/en/actions/using-workflows/reusing-workflows

Reply:
Yeah, it seemed like it relies on a particular naming scheme as well as a few other refactors I was hesitant to change in this PR. I think it might not be a bad change in the future to make, though.

Reply:
Can we remove the TODO for now? It seems like there are some unknowns with this effort that need to be ironed out.

Reply:
Is the new comment alright? I would like to still note that this is something we should look into to reduce redundant code.

Reply:
Maybe this is fine. I might be over-indexing here, but was just concerned a community member may come along and see the TODO and take some initiative to do the work and create a PR only for maintainers to not like the outcome. So it might be a bad experience for a contributor.